### ISSUE TITLE
ldap configuration does not persist on container replacement

### ISSUE THREAD

**Issue by tkent-xetus** [Graylog2/graylog2-images#65]
(This applies to the docker image)

Similar to #51 (thanks for fixing that, btw) the ldap configuration is blown away whenever the container is replaced. No ldap-backed user can login to the new container until a local admin re-configures the ldap settings.

Easy to plan for if you know it's going to happen, but it would be great if the settings persisted.

**Comment by joschi** [comment]
This is probably because the `password_secret` setting in the Graylog configuration file is regenerated every time the Docker image is being rebuilt.

The LDAP settings rely on the `password_secret` for encrypting the LDAP credentials.

**Comment by mariussturm** [comment]
@tkent-xetus you can persist the `password_secret` with `-e GRAYLOG_SERVER_SECRET=somesecretsaltstring`. Could you check if this change the behaviour? Keep the same string on every container.

**Comment by mariussturm** [comment]
Sorry can not reproduce this issue. Closing now, feel free to re-open if the problem persist.

**Comment by tkent-xetus** [comment]
Thanks for the quick response and sorry for not responding sonner. A few things.

First, you're completely right, setting a consistent `GRAYLOG_SERVER_SECRET` works. My fault for not realizing that was connected. This case should be closed.

That be said, I just spent a few hours today learning some really valuable lessons related to this that I should share here.

1. If you change the `GRAYLOG_SERVER_SECRET` (which you can), you also need to go into the mongodb in the container and remove any configuration that uses that salt. In this case I had to remove the `ldap_settings` collection record. If you don't remove this record, you'll be able to login and setup a new ldap configuration, only to have all users (including non-LDAP users) be locked out. There'll be an exception in reading the LDAP configuration that breaks all authentication. My fault for not thinking it through, but a very painful lesson.
2. Inputs created *before* the fix in #51 won't persist and you really want to clear them out of the `inputs` collection in the embedded mongodb. If you don't do that, strange error messages show up about inputs not being started.

BTW for other's reading this - just don't change the `GRAYLOG_SERVER_SECRET`. Even if you forgot to explicitly set it the first time, you can go find out what it is before replacing the container:

```
docker exec <your container> cat /opt/graylog/conf/graylog.conf \
   | grep password_secret | awk '{print  $3 }'
 ```

**Comment by loopix** [comment]
@tkent-xetus : thenks so much for your research! I just encountered this issue yesterday and I found your post in the train back to home. 

I confirm that the problem was exactly what you described and I was able to get back the previous secret and put it back to the new container to fix it.

Warning: at the first try, it failed ... because my ldap user has been locked due to too much login attempt : good to know !
